migrations/src/main/scala/za/co/absa/enceladus/migrations/continuous/EntityVersionMapMongo.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.migrations.continuous
+
+import org.bson.codecs.configuration.CodecRegistries.{fromProviders, fromRegistries}
+import org.bson.codecs.configuration.CodecRegistry
+import org.mongodb.scala.bson.codecs.DEFAULT_CODEC_REGISTRY
+import org.mongodb.scala.bson.codecs.Macros._
+import org.mongodb.scala.bson.collection.immutable.Document
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Filters._
+import org.mongodb.scala.model.IndexOptions
+import org.mongodb.scala.{MongoCollection, MongoDatabase}
+import za.co.absa.enceladus.migrations.continuous.model.EntityVersionMapping
+import za.co.absa.enceladus.migrations.framework.dao.ScalaMongoImplicits
+
+import scala.reflect.ClassTag
+
+final class EntityVersionMapMongo(db: MongoDatabase) extends EntityVersionMap {
+  private val codecRegistry: CodecRegistry = fromRegistries(fromProviders(classOf[EntityVersionMapping]),
+    DEFAULT_CODEC_REGISTRY)
+
+  import ScalaMongoImplicits._
+
+  private val entityMapCollectionName = "entity_version_map"
+
+  ensureCollectionExists()
+
+  /**
+    * An implementation if this abstract class should redefine this method. Clients of this class should use add()
```

Comment:
This is not abstract class. I'm starting to wonder if copying the scaladocs to subclasses has any value.